staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/metrics/metrics.go

```diff
@@ -32,24 +32,41 @@ const (
 )
 
 var (
-	encryptionConfigAutomaticReloadFailureTotal = metrics.NewCounterVec(
+	encryptionConfigAutomaticReloadsTotal = metrics.NewCounterVec(
 		&metrics.CounterOpts{
 			Namespace:      namespace,
 			Subsystem:      subsystem,
-			Name:           "automatic_reload_failures_total",
-			Help:           "Total number of failed automatic reloads of encryption configuration split by apiserver identity.",
+			Name:           "automatic_reloads_total",
+			Help:           "Total number of reload successes and failures of encryption configuration split by apiserver identity.",
 			StabilityLevel: metrics.ALPHA,
 		},
+		[]string{"status", "apiserver_id_hash"},
+	)
+
+	// deprecatedEncryptionConfigAutomaticReloadFailureTotal has been deprecated in 1.30.0
+	// use encryptionConfigAutomaticReloadsTotal instead
+	deprecatedEncryptionConfigAutomaticReloadFailureTotal = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Namespace:         namespace,
+			Subsystem:         subsystem,
+			Name:              "automatic_reload_failures_total",
+			Help:              "Total number of failed automatic reloads of encryption configuration split by apiserver identity.",
+			StabilityLevel:    metrics.ALPHA,
+			DeprecatedVersion: "1.30.0",
```

Comment:
Does this deprecation show up in a way that we can test?

Reply:
Thanks for adding that.  I find it a bit odd that you have to manually construct a fake registry at the right version just to see this message.

Reply:
Removed the new test and instead modified the existing ones to ensure `(Deprecated since 1.30.0)` exists (ref: https://github.com/kubernetes/kubernetes/compare/68b066017814efc0a252e31ab23e8c17fb3d4c2d..ce9e6b0d5fbdc840323f533c6001bf64f2b34238)